src/app/docs/file-attachment/file-attachment-docs.component.html

```diff
@@ -0,0 +1,189 @@
+
+    <sky-docs-demo-page-section
+      heading="Anatomy"
+    >
+      <sky-docs-anatomy
+        imagePath="~/assets/img/guidelines/file-attach/anatomy.png"
+      >
+        <sky-docs-anatomy-item>Option to upload local files</sky-docs-anatomy-item>
+        <sky-docs-anatomy-item>File name</sky-docs-anatomy-item>
+        <sky-docs-anatomy-item>File size</sky-docs-anatomy-item>
+        <sky-docs-anatomy-item>File preview</sky-docs-anatomy-item>
+        <sky-docs-anatomy-item>Delete button</sky-docs-anatomy-item>
+        <sky-docs-anatomy-item isOptional="true">Option to link to external files</sky-docs-anatomy-item>
+        <sky-docs-anatomy-item isOptional="true">User-entered metadata</sky-docs-anatomy-item>
+      </sky-docs-anatomy>
+    </sky-docs-demo-page-section>
+
+    <sky-docs-demo-page-section
+      heading="Options"
+    >
+      <sky-docs-design-guideline
+        heading="Link to external files"
+      >
+        <p class="sky-docs-text-block">
+          You can indicate whether to include the option to attach external files using their URLs.
+        </p>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline
+        heading="Metadata"
+      >
+        <p class="sky-dpcs-text-block">
+          You can include inputs for users to add metadata about file attachments.
+        </p>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline
+        heading="Validation"
+      >
+        <p class="sky-docs-text-block">
+          You can specify parameters for valid files, including file types, minimum and maximum sizes, and custom validation.
+        </p>
+      </sky-docs-design-guideline>
+    </sky-docs-demo-page-section>
+
+    <sky-docs-demo-page-section
+      heading="Behavior & States"
+    >
+      <sky-docs-design-guideline
+        heading="File information"
+      >
+        <p>
+          After users attach files, summary information appears below the file attachment options. For local files, the default summary includes the file name, file size, file preview, and a delete button. For external files, the default summary includes the URL and a delete button. You can include additional inputs to display user-entered metadata.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/file-attach/behavior-preview.png"
+        >
+        </sky-docs-design-guideline-thumbnail>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline
+        heading="Responsiveness"
+      >
+        <p>
+          The file attachment element switches to a vertical layout in smaller viewports.
+        </p>
+        
+        <sky-docs-design-guideline-thumbnail
+          imageSource="~/assets/img/guidelines/file-attach/behavior-responsive.png"
+        >
+        </sky-docs-design-guideline-thumbnail>
+      </sky-docs-design-guideline>
+    </sky-docs-demo-page-section>
+
+    <sky-docs-demo-page-section
```

Comment:
Based on the existing docs, `sky-docs-demo-page-section` should be `sky-docs-design-guideline` because this is a subsection of behavior and states, not it's own page anchor-level section.

Reply:
I used that option because when I made it a guideline, it forced all of the images into a really small size within the table that made it a little difficult to see. I can make the switch though

Reply:
I guess we should leave this as is for now, but long-term it shouldn't be like this since this is a subsection of the *Behavior and states* section.

@Blackbaud-SteveBrush, is the behavior Denise was seeing caused by https://github.com/blackbaud/skyux-docs-tools/issues/33?

Reply:
That's not really an apples-to-apples comparison. In the dropdown docs, that *Options* section is a top-level section within the Design tab (i.e., a page anchor created by a `sky-docs-design-guideline` element). To make the table work here, Denise made the *States* subsection a top-level section too even though it is supposed to be a subsection of the *Behavior and states* section.

It seems like the table format works when it is within a top-level section on the Design tab (i.e., a page anchor created by a `sky-docs-design-guideline` element), but the table format seems to blow up when it is in a subsection (i.e., not a page anchor but a child section created with a `sky-docs-demo-page-section` element). I have no idea how difficult it would be to fix this, and maybe the solution is to reconsider how we write Design guidelines so that we only include tables in top-level items. This would avoid fixing the table to work wherever we want it, but it creates an odd workflow and won't be clear to anyone who isn't aware of the `docs-tools` limitation.

Anyway, hopefully that's all clear, but let me know if we should talk through this more.

Reply:
Should I merge this and file an issue to get back to this or....?

Reply:
@Blackbaud-SteveBrush, should we file an issue to update docs-tools to make the table styles work in `sky-docs-demo-page-section` (or to at least research what's going on)? Or should we just consider it an intentional limitation that the table styles only works with `sky-docs-design-guideline`?

Reply:
I'm going to clone this branch locally to see if it's a quick fix.

Reply:
Got a PR to address this. Instead of `sky-docs-design-guideline-thumbnail` we would use the new `sky-docs-thumbnail` to be used in tables. https://github.com/blackbaud/skyux-docs-tools/pull/106

Reply:
(We would still use `sky-docs-design-guideline-thumbnail` if we want a left and right-hand column.)

Reply:
Should be good now.

Reply:
I saw! Yay, thanks so much, Steve!